[BE] 회원가입 API 기능

bohyeon-n opened the pull request:
- 회원가입 API 구현 
- 중복 userID 체크 API 구현 
- 중복 mobile 체크 API 구현 
- /logs 폴더 ignore 하도록 .gitignore 파일 수정함 
- 기존에 올라와있던 /logs 폴더 삭제함

bohyeon-n committed:
회원 가입 기능 구현

회원 가입 폼 전송 시 null 체크 후 회원 추가함

bohyeon-n committed:
.gitignore 파일에 /logs 폴더 추가함

/logs는 프로젝트에 올리지 않기 위해 .gitignore 파일 수정함

bohyeon-n committed:
중복 체크 api 기능 추가

userID와 mobile 중복 체크 api 기능 추가함

realsigridjin commented on BE-workspace/src/main/java/com/codesquad/signup4/controller/userapi/APIUserCreateController.java:
```diff
@@ -7,58 +7,64 @@
-      throw new BadRequestException();
+    @InitBinder
+    public void initBinder(WebDataBinder binder) {
+        binder.initDirectFieldAccess();
```
여기서 initBinder를 사용해서 initDirectFieldAccess() 로 바로 필드 주입을 시도하신 이유가 있나요? create 메소드는 어짜피 POST 요청이라 바로 필드에 값을 주입하는 것으로 알고 있습니다.

bohyeon-n replied:
setter가 없는 필드는 받아오지 못해서 InitBinder를 추가해주면 된다고 해서 이렇게 구현했는데요~ 
바로 필드에 값을 주입한다는 말이 무슨말인가요?

realsigridjin replied:
https://jojoldu.tistory.com/407 여기 글을 보면 setter가 없는 필드에도 스프링 부트에 내장된 Jackson 라이브러리가 값을 주입해준다고 나와 있어요.

realsigridjin replied:
저도 이번에 initBinder를 처음 봐서요.. 저걸 달아주면 컨트롤러에서 setter가 없는 필드(유저) 에게 값을 주입해주나요? 그냥 setter 메소드를 만들어주는 것과 비교해서 더 좋은 점이 있나요?

bohyeon-n replied:
`@RequestBody`쓰면 dto 객체를 쓴다는 의미여서 ,dto 객체로 테스트 해봤는데 안되네요 ~ 검색해보니, '@RequestBody`쓰지 말고 `@ModelAttribute`쓰라고 해서 해봤는데 이것도 안돼요... 
일단 이렇게 하고 추후에 더 공부해서 수정할게요!

realsigridjin commented on BE-workspace/src/main/java/com/codesquad/signup4/domain/UserRepository.java:
```diff
@@ -7,4 +7,7 @@
   @Query("SELECT * FROM USER u where u.user_iD = :userID")
   User findByUserID(@Param("userID") String userID);
+
+  @Query("SELECT * FROM USER u where u.mobile = :mobile")
```
직접 쿼리문을 만들어서 검색하도록 설계하셨군요. 좋은 것 같습니다.

realsigridjin commented on BE-workspace/src/main/java/com/codesquad/signup4/controller/userapi/APIUserCreateController.java:
```diff
@@ -7,58 +7,64 @@
-      throw new BadRequestException();
-    }
+    @PostMapping("/create")
+    public Result create(User newUser) {
```
레포지토리에 실제 사용자 정보를 저장하는 로직이 보이지 않습니다.

sigridjineth committed:
chore: 중복 이메일 체크 API 기능 추가
- UserRepository에 query 를 입력하여 이메일 반환 메소드 구현
- 컨트롤러에서 Result.fail 또는 Ok를 리턴하도록 구현

jypthemiracle commented:
중복 이메일 체크 API 기능을 엘리님과 유사한 구조로 구현하였습니다.

sigridjineth committed:
chore: 중복 아이디, 이메일, 핸드폰 번호를 체크 로직을 중복 제거함

jypthemiracle commented:
중복 아이디, 이메일, 휴대폰 번호를 체크하는 로직이 중복되어 이를 통합할 수 있는 메소드를 만들었습니다.

bohyeon-n committed:
create 메서드 수정

user를 저장하는 코드가 빠져있어서 추가함

bohyeon-n committed:
Merge branch 'BE/feature/32/join' of https://github.com/codesquad-memeber-2020/signup-4 into BE/feature/32/join

# Conflicts:
#	BE-workspace/src/main/java/com/codesquad/signup4/controller/userapi/APIUserCreateController.java
